CMakeLists.txt

```diff
@@ -60,11 +60,29 @@ IF(MSVC)
     ADD_COMPILE_FLAG("/arch:sse2")
   ENDIF()
   ADD_COMPILE_FLAG("/wd4146") # Ignore warning "warning C4146: unary minus operator applied to unsigned type, result still unsigned", this pattern is used somewhat commonly in the code.
+  # 4267 and 4244 are conversion/truncation warnings. We might want to fix these but they are currently pervasive.
+  ADD_COMPILE_FLAG("/wd4267")
+  ADD_COMPILE_FLAG("/wd4244")
   ADD_COMPILE_FLAG("/WX-")
-  ADD_DEBUG_COMPILE_FLAG("/O0")
+  ADD_DEBUG_COMPILE_FLAG("/Od")
   ADD_NONDEBUG_COMPILE_FLAG("/O2")
-  ADD_COMPILE_FLAG("-D_CRT_SECURE_NO_WARNINGS")
-  ADD_COMPILE_FLAG("-D_SCL_SECURE_NO_WARNINGS")
+  ADD_COMPILE_FLAG("/D_CRT_SECURE_NO_WARNINGS")
+  ADD_COMPILE_FLAG("/D_SCL_SECURE_NO_WARNINGS")
+
+  ADD_NONDEBUG_COMPILE_FLAG("/UNDEBUG") # Keep asserts.
+  # Also remove /D NDEBUG to avoid MSVC warnings about conflicting defines.
+  if( NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
+    foreach (flags_var_to_scrub
```

Comment:
isn't this loop what `ADD_NONDEBUG_COMPILE_FLAG` does? (I don't know cmake well, but superficially it looks similar). if not, maybe a utility function alongside that one?

otherwise lgtm but i know almost nothing about windows.